Source/Core/AudioCommon/Mixer.h

```diff
@@ -54,16 +55,102 @@ class Mixer final
   static constexpr u64 FIXED_SAMPLE_RATE_DIVIDEND = 54000000 * 2;
 
 private:
-  static constexpr u32 MAX_SAMPLES = 1024 * 4;  // 128 ms
-  static constexpr u32 INDEX_MASK = MAX_SAMPLES * 2 - 1;
-  static constexpr int MAX_FREQ_SHIFT = 200;  // Per 32000 Hz
-  static constexpr float CONTROL_FACTOR = 0.2f;
-  static constexpr u32 CONTROL_AVG = 32;  // In freq_shift per FIFO size offset
-
   const unsigned int SURROUND_CHANNELS = 6;
 
   class MixerFifo final
   {
+    static constexpr size_t GRANULE_QUEUE_SIZE = 20;
+
+    /**
+     * import numpy as np
+     * import scipy.signal as signal
+     * window = np.convolve(np.ones(128), signal.windows.dpss(128 + 1, 3))
+     * window /= (window[:len(window) // 2] + window[len(window) // 2:]).max()
+     * window = np.round(window * (2 ** 16))
+     * elements = ", ".join([f"{int(x)}" for x in window])
+     * print(f'static constexpr std::array<int, {len(window)}> GRANULE_WINDOW = {{ {elements} }};')
+     */
```

Comment:
Our [style guide](https://github.com/dolphin-emu/dolphin/blob/master/Contributing.md) says to use single-line comments instead of multi-line.